### PR TITLE
Stunner - New Marshallers: add support to multi instance subprocess

### PR DESCRIPTION
**Pull request by evacchi**
@romartin @hasys please take a look

### CODE REVIEW

**Comment by evacchi** [comment]
@romartin I don't know where/if the tests have actually failed here...

**Comment by evacchi** [comment]
ping @hasys

**Comment by hasys** [comment]
Jenkins retest please

**Comment by evacchi** [comment]
PING @hasys @romartin I don't think failing tests are related

**Comment by hasys** [comment]
@evacchi, ok but I am still need build to test it.

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
@hasys can you take a look please ?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/fromstunner/properties/MultipleInstanceSubProcessPropertyWriter.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    public void setTrigger(String miTrigger) {
+
+    }
+
+    public void setCompletionCondition(String expression) {
+        FormalExpression formalExpression = bpmn2.createFormalExpression();
+        formalExpression.setBody(expression);
+        this.miloop.setCompletionCondition(formalExpression);
+
+        // BLAHHH
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/fromstunner/properties/MultipleInstanceSubProcessPropertyWriter.java]: BLAHHH? :)

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/fromstunner/properties/MultipleInstanceSubProcessPropertyWriter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        dataOutput.setItemSubjectRef(item);
+        this.ioSpec.getDataOutputs().add(dataOutput);
+        OutputSet outputSet = bpmn2.createOutputSet();
+        this.ioSpec.getOutputSets().add(outputSet);
+        outputSet.getDataOutputRefs().add(dataOutput);
+        this.miloop.setLoopDataOutputRef(dataOutput);
+        return dataOutput;
+    }
+
+    public void setTrigger(String miTrigger) {
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/fromstunner/properties/MultipleInstanceSubProcessPropertyWriter.java]: Is this method should be empty?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/MultipleInstanceSubProcessPropertyReader.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public String getCollectionInput() {
+        MultiInstanceLoopCharacteristics loopCharacteristics = getMultiInstanceLoopCharacteristics();
+        ItemAwareElement ieDataInput = loopCharacteristics.getLoopDataInputRef();
+        return process.getDataInputAssociations().stream()
+                .filter(dia -> dia.getTargetRef().equals(ieDataInput))
+                .map(dia -> dia.getSourceRef().get(0).getId())
+                .findFirst()
+                .get();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/MultipleInstanceSubProcessPropertyReader.java]: Is it safe to call `get()` without `isPresent()` there?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/MultipleInstanceSubProcessPropertyReader.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public String getCollectionOutput() {
+        MultiInstanceLoopCharacteristics loopCharacteristics = getMultiInstanceLoopCharacteristics();
+        ItemAwareElement ieDataOutput = loopCharacteristics.getLoopDataOutputRef();
+        return process.getDataOutputAssociations().stream()
+                .filter(doa -> doa.getSourceRef().get(0).equals(ieDataOutput))
+                .map(doa -> doa.getTargetRef().getId())
+                .findFirst()
+                .get();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/main/java/org/kie/workbench/common/stunner/bpmn/backend/converters/tostunner/properties/MultipleInstanceSubProcessPropertyReader.java]: Same question here: Is it safe to call get() without isPresent() there?

**Comment by evacchi** [comment]
> Sequence flow NullPointer
This is related to PR #1779 

I think I have fixed most of the other issues now, added tests. @hasys

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
@hasys I think that this is ready to go now

Jenkins execute full downstream build

**Comment by hasys** [comment]
@evacchi same here, sorry but build is removed. Let's try again:

Jenkins execute full downstream build

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
error unrelated, but still.

Jenkins execute full downstream build

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
build successful. @hasys ? @romartin ?

**Comment by evacchi** [comment]
@hasys ping ?

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by evacchi** [comment]
Jenkins please retest this

**Comment by evacchi** [comment]
Jenkins please retest this

**Comment by evacchi** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Hey @evacchi , it seems J is still aborting the build... fired a rebuild, plz keep an eye on this and lemme know when ready to merge, thanks!

**Comment by evacchi** [comment]
Jenkins execute full downstream build

**Comment by evacchi** [comment]
artifacts generated. LGTM @romartin

**Pull request merged**